packages/strapi-plugin-content-type-builder/admin/src/containers/AttributeForm/index.js

```diff
@@ -38,9 +38,25 @@ class AttributeForm extends React.Component {
   state = { didCheckErrors: false, formErrors: {}, showForm: false };
 
   getCurrentForm = () => {
-    const { activeTab, attributeType } = this.props;
+    const { activeTab, attributeType, modifiedData } = this.props;
+    const isRepeatable = get(modifiedData, 'repeatable', false);
+    const form = get(
+      supportedAttributes,
+      [attributeType, activeTab, 'items'],
+      []
+    );
+
+    if (
```

Comment:
It would be better to convert your supportAttributes.json to js and support functions so you can implement this logic in the supportedAttributes directly. Sth like:

```js
const attributeConfig = get(supportedAttributes, [attributeType, activeTab]);
const items = typeof attributes === 'function' ? attributeConfig(modifiedData) : attributeConfig;
```

and  in supportedAttributes.js
```js
module.exports = {
//...
group: {
  //...
  advanced: (data) => {
     const items = [requiredFiel];
     if (data.repeatable === true) {
        return items.concat(minItems,maxItems);
     }
     return items; 
  }
}
```

Reply:
Agree!